File: Daily Test/Java/Count Common Set Bits 8.9.2025.java
import java.util.*;

public class Main {
    static int countBits(int x) {
        int cnt = 0;
        while (x > 0) {
            cnt += (x & 1);
            x >>= 1;
        }
        return cnt;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) arr[i] = sc.nextInt();

        int res = arr[0];
        for (int i = 1; i < n; i++) res &= arr[i];

        System.out.println(countBits(res));
    }
}

File: Daily Test/C/Count Common Set Bits 8.9.2025.c
#include <stdio.h>

int countBits(int x) {
    int cnt = 0;
    while (x) {
        cnt += (x & 1);
        x >>= 1;
    }
    return cnt;
}

int main() {
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);

    int res = arr[0];
    for (int i = 1; i < n; i++) res &= arr[i];

    printf("%d\n", countBits(res));
    return 0;
}